packages/calcite-components/src/components/modal/modal.tsx

```diff
@@ -532,7 +529,8 @@ export class Modal
     this.open = false;
   };
 
-  private openModal() {
+  private async openModal(): Promise<void> {
+    await componentOnReady(this.el);
```

Comment:
@jcfranco is there any way we could put this in `activateFocusTrap` and/or `connectFocusTrap` so that this would happen for any component using focus trapping?

Reply:
It might not be that helpful since the main issue is related to watchers and refs. `focusTrapComponent` utils seem disconnected from this, right?

Reply:
Yeah that's true. I was just thinking that this problem might arise in other components as well.

Reply:
Agreed. Maybe we could wire it up if we made some additional changes to `focusTrapComponent`. 🤔❓